src/python/pants/backend/python/lint/pylint/rules.py

```diff
@@ -194,57 +178,70 @@ async def pylint_lint_partition(
     )
 
 
-@rule(desc="Lint using Pylint", level=LogLevel.DEBUG)
-async def pylint_lint(
-    request: PylintRequest,
-    pylint: Pylint,
-    python_setup: PythonSetup,
-    first_party_plugins: PylintFirstPartyPlugins,
-) -> LintResults:
-    if pylint.skip:
-        return LintResults([], linter_name=request.name)
-
-    # Pylint needs direct dependencies in the chroot to ensure that imports are valid. However, it
-    # doesn't lint those direct dependencies nor does it care about transitive dependencies.
-    linted_targets = await Get(
-        Targets, Addresses(field_set.address for field_set in request.field_sets)
-    )
+# TODO(#10863): Improve the performance of this, especially by not needing to calculate transitive
+#  targets per field set. Doing that would require changing how we calculate interpreter
+#  constraints to be more like how we determine resolves, i.e. only inspecting the root target
+#  (and later validating the closure is compatible).
```

Comment:
> Consider using CoarsenedTarget to represent these

Hm does that have the semantics we want for Python import cycles, which are legal? That seems better as a dedicated followup change.

> you won't need to worry about breaking out a separate @rule for performance reasons.

This was for ease of testing, not performance. Re: performance, my point was it's nice it will now show up in benchmarks. Nothing else.

Reply:
> Hm does that have the semantics we want for Python import cycles, which are legal? That seems better as a dedicated followup change.

It does, yea. But fine as a followup change.